Dockerfile

```diff
@@ -0,0 +1,48 @@
+# syntax=docker/dockerfile:1.4
+
+# Use rust:bookworm as the build environment
+FROM rust:bookworm as build-environment
+
+# Set non-interactive mode for apt so it doesn't ask for user input during the build
+ENV DEBIAN_FRONTEND=noninteractive
+
+# # Install required dependencies
+# RUN apt-get update && apt-get install -y clang lld curl build-essential linux-generic git \
+#     && curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs > rustup.sh \
+#     && chmod +x ./rustup.sh \
+#     && ./rustup.sh -y
+
+RUN apt-get update && apt-get install cmake -y
+    
+ARG TARGETARCH
+WORKDIR /opt
+
+
+# Add the required CFLAGS if the TARGETARCH matches
+RUN [[ "$TARGETARCH" = "x86_64-unknown-linux-gnu" ]] && echo "export CFLAGS=-mno-outline-atomics" >> $HOME/.profile || true
+
+WORKDIR /opt/cryo
+COPY . .
```

Comment:
to avoid pulling unnecessary parts of repo I would explicitly add `crates` and `Cargo.toml` `Cargo.lock` are src copy